mix.exs

```diff
@@ -30,6 +30,7 @@ defmodule Coap.MixProject do
       {:ex_doc, "~> 0.23", only: :dev, runtime: false},
 
       # Runtime
+      {:gen_state_machine, "~> 3.0"},
```

Comment:
I haven't used this before. What do we feel it brings to the table? I'm generally pretty conservative about adding new dependencies.

Reply:
GenStateMachine is just a pure elixir wrapper around OTP's `gen_statem` (rewrite of `gen_fsm` since OTP 18 or 19).
I understand your concern about a (possible unstable) dependancy. I see no problem using directly OTP's `gen_statem`. It should just add some extra lines of code.

Reply:
This sounds fine. Wrappers usually stay pretty stable, and I'm happy to use the elixir over the OTP directly.